component/src/main/java/com/celements/photo/utilities/ImageUrlExtractor.java

```diff
@@ -0,0 +1,91 @@
+package com.celements.photo.utilities;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.celements.model.context.ModelContext;
+import com.celements.photo.container.ImageUrlDim;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.web.Utils;
+
+public class ImageUrlExtractor {
+
+  public static final Logger LOGGER = LoggerFactory.getLogger(ImageUrlExtractor.class);
+
+  public static int MIN_SOCIAL_MEDIA_IMAGE_SIZE = 200;
+  public static int MIN_SOCIAL_MEDIA_AREA_SIZE = MIN_SOCIAL_MEDIA_IMAGE_SIZE
+      * MIN_SOCIAL_MEDIA_IMAGE_SIZE;
+
+  public List<ImageUrlDim> extractImagesList(String content) {
+    String regex = "<img .*?src=['\"](.*?)['\"].*?/>";
+    Map<Long, List<ImageUrlDim>> articleImages = new TreeMap<>();
+    Pattern p = Pattern.compile(regex);
```

Comment:
The regex compilation is expensive. Thus move the Regex and its combiled pattern to a constant. The Pattern class is thread safe to support this use: http://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html